integration_tests/test_snapcraft_recording.py

```diff
@@ -197,3 +198,108 @@ def test_prime_with_packages_missing_dependency(self):
         self.assertEqual(
             recorded_yaml['parts'][part_name][self.packages_type],
             expected_packages)
+
+
+class SnapcraftRecordingBzrSourceTestCase(
+        integration_tests.BzrSourceBaseTestCase,
+        SnapcraftRecordingBaseTestCase):
+
+    def test_prime_with_bzr_source(self):
+        self.copy_project_to_cwd('bzr-head')
+
+        self.init_and_config_bzr()
+        self.commit('"test-commit"', unchanged=True)
+
+        self.run_snapcraft('prime')
+
+        recorded_yaml_path = os.path.join(
+            self.prime_dir, 'snap', 'snapcraft.yaml')
+        with open(recorded_yaml_path) as recorded_yaml_file:
+            recorded_yaml = yaml.load(recorded_yaml_file)
+
+        commit = self.get_revno()
+        self.assertEqual(
+            recorded_yaml['parts']['bzr']['source-commit'], commit)
+
+
+class SnapcraftRecordingGitSourceTestCase(
+        integration_tests.GitSourceBaseTestCase,
+        SnapcraftRecordingBaseTestCase):
+
+    def test_prime_with_git_source(self):
+        self.copy_project_to_cwd('git-head')
+
+        self.init_and_config_git()
+        self.commit('"test-commit"', allow_empty=True)
+
+        self.run_snapcraft('prime')
+
+        recorded_yaml_path = os.path.join(
+            self.prime_dir, 'snap', 'snapcraft.yaml')
+        with open(recorded_yaml_path) as recorded_yaml_file:
+            recorded_yaml = yaml.load(recorded_yaml_file)
+
+        commit = self.get_revno()
+        self.assertEqual(
+            recorded_yaml['parts']['git']['source-commit'], commit)
+
+
+class SnapcraftRecordingHgSourceTestCase(
+        integration_tests.HgSourceBaseTestCase,
+        SnapcraftRecordingBaseTestCase):
+
+    def test_prime_with_hg_source(self):
+        self.copy_project_to_cwd('hg-head')
+
+        self.init_hg()
+        open('1', 'w').close()
+        self.commit('1', '1')
+
+        self.run_snapcraft('prime')
+
+        recorded_yaml_path = os.path.join(
+            self.prime_dir, 'snap', 'snapcraft.yaml')
+        with open(recorded_yaml_path) as recorded_yaml_file:
+            recorded_yaml = yaml.load(recorded_yaml_file)
+
+        commit = self.get_id()
+        self.assertEqual(
+            recorded_yaml['parts']['mercurial']['source-commit'], commit)
+
+
+class SnapcraftRecordingSubversionSourceTestCase(
+        integration_tests.SubversionSourceBaseTestCase,
+        SnapcraftRecordingBaseTestCase):
+
+    def test_prime_with_subversion_source(self):
+        self.copy_project_to_cwd('svn-pull')
+
+        self.init_svn()
+
+        subprocess.check_call(
+            ['svn', 'checkout',
+             'file:///{}'.format(os.path.join(self.path, 'repo')),
+             'local'],
+            stdout=subprocess.DEVNULL)
+
+        open(os.path.join('local', 'file'), 'w').close()
+        subprocess.check_call(
+            ['svn', 'add', 'file'], stdout=subprocess.DEVNULL, cwd='local/')
```

Comment:
Yes, I'll move it. But ugh, svn is so ugly. I will have to add an argument for the cwd.